### PR TITLE
Allow for "prerelease" version bump

### PR DESCRIPTION
**Pull request by scotttrinh**
I wasn't an expert on `npm version`, but I guess I kinda am now 😳 

In order to bump the prerelease without changing the main part of the version number, I need to specify the version bump as `prerelease`. The last change I made in #21 just allows setting `premajor`, `preminor`, and `prepatch`.

### CODE REVIEW

**Comment by scotttrinh** [comment]
Actually I'm going to close this and update the interface to be more explicit.

**Comment by scotttrinh** [comment]
Instead of mediating the options to `npm version` and `npm publish`, we're now passing them through a bit more directly. This allows us to do any kind of version bumping allowed by `npm version` and makes things like `prerelease` a little easier.

File: bin/release [hunk 1]
```diff
@@ -37,19 +35,13 @@ main() {
...
-  if [[ "$prerelease_flag" == "--prerelease" ]]; then
-    git_hash=$(git rev-parse --verify HEAD --short=8)
-    npm version pre${release_type} --preid $git_hash
-    npm publish --tag next
-  else
-    npm version $release_type
-    npm publish
-  fi
+  # Add a preid of "rc" for the varirous prerelease version
+  npm version ${release_type} --preid rc
```
Comment (mendrew) [inline bin/release:43]: > --preid - Identifier to be used to prefix premajor, preminor,
        prepatch or prerelease version increments.


So, this `--preid` doesn't throw any errors or adds `rc` when we use it with the default 'latest', right?

If so, very nice!

**Review by mendrew — APPROVED** [review]
Looks good!

One question.

File: bin/release (hunk 1 above)
Comment (scotttrinh) [reply bin/release:43]: It doesn't throw errors, but it _would_ add `rc.0` if you used `premajor|preminor|prepatch|prerelease` as your version string but then didn't add `next` as your tag. Let me add some extra explanation and examples to the README.

Comment (mendrew) [reply bin/release:43]: Cool, thank you for the Readme update!!! Clear!

**Review by mendrew — APPROVED** [review]

**Pull request merged**